File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Go/s826491904_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	N, _ := strconv.Atoi(readLine(reader))
	max, min, _ := strconv.Atoi(readLine(reader))
	for i := 1; i < N; i++ {
		num, _ := strconv.Atoi(readLine(reader))
		if num > max {
			max = num
		}
		if num < min {
			min = num
		}
	}
	fmt.Println(max - min)
}

func readLine(reader *bufio.Reader) string {
	line, _ := reader.ReadString('\n')
	return strings.TrimSpace(line)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Python/s826491904_commented.py
import sys

def solve():
    input = sys.stdin.read
    data = input().split()
    
    N = int(data[0])
    max_val = min_val = int(data[1])
    
    for i in range(2, N + 1):
        num = int(data[i])
        max_val = max(max_val, num)
        min_val = min(min_val, num)
        
    print(max_val - min_val)

solve()  # End of Code
